torch/csrc/jit/OVERVIEW.md

```diff
@@ -886,267 +884,244 @@ def LSTMCellS(x, hx, cx, w_ih, w_hh, b_ih, b_hh):
+
+`RemoveExpands` removes implicitly inserted (by tracing) expand nodes which might not necessarily be always valid when used inside script methods that might have unstable shapes.
+
+`Dead Code Elimination` (DCE) removes the expressions that are not needed to compute the results and do not have side effects either.
+
+`Common Subexpression Elimination` (CSE) looks for identical expressions and reuses one of the expressions in lieu of the rest. 
+
+`Constant Propagation` recognizes and evaluates constant expressions at compile-time, so these expressions don't have to be recomputed every time a scripted function is executed.
+
+Peephole performs a variety of algebraic simplifications. For example, it recognizes that adding 0 or multiplying by 1 is an identity operation, so is `a.t().t()`. 
+
+The passes above are also *derivative preserving*. If a computation the graph `requires_grad` and it is valid to compute its derivative, then these passes are only allow to replace that computation with another computation that is also differentiable. In other words, these passes cannot break the ability for autograd to work correctly. Algebraic rewrites and peephole optimizations are generally derivative preserving but something 
+
+*Post-derivative optimization* The next optimizations depends on whether any part of the graph actual requires a gradient to be calculated, which is determined by `needsGradient`. In the case where no gradients are required (i.e. for inference graphs), then we can directly apply optimizations that generate graphs that may not have valid gradients defined. Before those optimizations are performed, the executor runs `RemoveProfileNodesAndSpecializeTypes` to move the profiled types onto the output `Value`s and remove `prim::profile` nodes from the graph. This optimization greatly reduces the complexity of `BatchMM` and `FuseTensorExprs` passes that are performed next.
+
+`BatchMM` performs two separate optimizations. The first one looks for `aten::add`s whose arguments are matrix multiplies and replaces them with one `aten::mm` of two concats which concatenate the LHSs of both `aten::mm`s and the RHSs, respectively. The second optimization looks for the matrix multiplies that use the same LHS or RHS and concatenates all the arguments on the opposite side. The result is then split back with `aten::chunk` into individual matrices.
+
+`FuseTensorExprs` looks for adjacent point-wise operations along with reviewing operations such as `split` and `concat`, and creates `prim::TensorExprGroup` Nodes in the graph to replace these operations. The Operator registered to execute `prim::TensorExprGroup` nodes will generate a new CUDA kernel for each unique Node, which replaces the original separate execution. 
```

Comment:
Fuser supports (conditionally) reductions as well - it's probably worth to not be too specific about what kind of ops are supported right now. Similarly, it is capable of generating kernels for CPU fusion groups too, not only for CUDA.